arangod/Replication2/ReplicatedLog/Components/StorageManager.cpp

```diff
@@ -324,7 +320,41 @@ auto StorageManager::getTermIndexMapping() const -> TermIndexMapping {
 
 auto StorageManager::getPeristedLogIterator(LogIndex first) const
     -> std::unique_ptr<PersistedLogIterator> {
-  return guardedData.getLockedGuard()->methods->read(first);
+  return getPeristedLogIterator(
+      LogRange{first, LogIndex{static_cast<std::uint64_t>(-1)}});
+}
+
+auto StorageManager::getPeristedLogIterator(std::optional<LogRange> bounds)
+    const -> std::unique_ptr<PersistedLogIterator> {
+  auto range =
+      bounds ? *bounds
+             : LogRange{LogIndex{0}, LogIndex{static_cast<std::uint64_t>(-1)}};
+
+  auto diskIter = guardedData.getLockedGuard()->methods->read(range.from);
+
+  struct Iterator : PersistedLogIterator {
+    explicit Iterator(LogRange range,
+                      std::unique_ptr<PersistedLogIterator> disk)
+        : _range(range), _disk(std::move(disk)) {}
+
+    auto next() -> std::optional<PersistingLogEntry> override {
+      while (true) {
```

Comment:
NIT. It would not matter for the correctness of the code, but is there any reason why this is wrapped inside a `while (true)`?

Reply:
copy pasta! thanks